File: Tracks/csharp/phone-number/PhoneNumber.cs
using System;
using System.Text.RegularExpressions;
using System.Linq;

public class PhoneNumber
{
    public static string Clean(string phoneNumber)
    {
        Match matchedPhoneNumber = Regex.Match(Regex.Replace(phoneNumber, @"\D", ""), @"^1?([2-9]\d{2}[2-9]\d{6})$");
        return matchedPhoneNumber.Success ? matchedPhoneNumber.Groups[1].ToString() : throw new ArgumentException();
    }
}

File: Tracks/typescript/phone-number/phone-number.ts
export default class PhoneNumber {
  parsedNumber: string;
  constructor(input: string) {
    this.parsedNumber = input;
  }

  number = () => {
    if (this.parsedNumber.match(/[a-zA-Z]/g) !== null) return undefined;
    const digits = (this.parsedNumber.match(/\d/g) || []).join("");
    const match = digits.match(/^1?(?<number>([2-9]{1}\d{2}){2}\d{4})$/);
    return (
      (match !== null && !!match.groups && match.groups["number"]) || undefined
    );
  };
}
